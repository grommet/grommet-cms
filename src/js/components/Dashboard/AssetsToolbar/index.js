/* @flow */
import React from 'react';
import Button from 'grommet/components/Button';
import Box from 'grommet/components/Box';
import TrashIcon from 'grommet/components/icons/base/Trash';
import CheckmarkIcon from 'grommet/components/icons/base/Checkmark';
import Toolbar from './toolbar';

type Props = {
  onClick: (e: Event) => void,
  buttonType?: 'DELETE' | 'SUBMIT'
}

export default function CmsToolbar({
  onClick,
  buttonType
}: Props): React$Element<*> {
  const type = buttonType || 'SUBMIT';
  return (
    <Toolbar>
      <Box colorIndex="light-2" pad="medium" align="end">
        <Button
          icon={type === 'SUBMIT' ? <CheckmarkIcon /> : <TrashIcon />}
          primary={type === 'SUBMIT'}
          className={type === 'DELETE' ? 'grommetux-button--critical' : ''}
          label={`${type === 'DELETE' ? 'delete' : 'use'} selected assets`}
          onClick={onClick}
        />
      </Box>
    </Toolbar>
  );
}

