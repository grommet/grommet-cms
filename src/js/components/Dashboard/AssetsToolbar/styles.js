import { css } from 'styled-components';

export default css`
  position: fixed;
  padding: 0px !important;
  left: 0;
  bottom: 0;
  width: 100%;
  z-index: 5;
  margin: 0 !important;
`;
