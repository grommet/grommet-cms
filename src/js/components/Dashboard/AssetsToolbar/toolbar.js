import styled from 'styled-components';
import styles from './styles';

export default styled.div`
  ${styles}
`;
