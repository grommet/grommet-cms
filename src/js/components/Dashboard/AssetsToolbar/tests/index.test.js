import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Button from 'grommet/components/Button';
import Box from 'grommet/components/Box';
import Toolbar from '../toolbar';
import AssetsToolbar from '../index';

describe('<AssetsToolbar />', () => {
  const wrapper = shallow(
    <AssetsToolbar onClick={jest.fn()} />
  );

  it('should create a snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should have a Toolbar component', () => {
    expect(wrapper.find(Toolbar).length).toBe(1);
  });

  it('should have a Button component', () => {
    expect(wrapper.find(Button).length).toBe(1);
  });

  it('should have a Box component', () => {
    expect(wrapper.find(Box).length).toBe(1);
  });
});
