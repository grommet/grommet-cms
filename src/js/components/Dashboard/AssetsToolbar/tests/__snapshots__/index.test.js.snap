exports[`<AssetsToolbar /> should create a snapshot 1`] = `
<styled.div>
  <Box
    align="end"
    announce={false}
    colorIndex="light-2"
    direction="column"
    focusable={true}
    pad="medium"
    responsive={true}
    tag="div">
    <Button
      label="use selected images"
      method="push"
      onClick={[Function]}
      primary={true}
      type="button" />
  </Box>
</styled.div>
`;
